File: solutions_5631989306621952_1/Python/Pay/word.py
ti = 0
tn = input()
while ti < tn:

  s = raw_input()
  a = s[0]
  s = s[1:]

  for x in s:
    if a[0] <= x:
      a = x + a
    else:
      a = a + x

  print 'Case #%s: %s' % (ti + 1, a)
  ti += 1

File: solutions_5631989306621952_1/C++/kangkang1990/word.cpp
#include <fstream>
#include <cstring>
#include <iostream>

using namespace std;

int main() {
	//set input/output stream
	ifstream input;
	ofstream output;
	input.open("input.txt");
	output.open("output.txt");

	int T;
	char str[2000];
	char outputStr[4000];

	input >> T;
	for(int x = 1; x <= T; x++) {
		input >> str;
		int len = strlen(str);
		output << "Case #" << x << ": ";
		int start = 2000;
		int end = 2000;
		outputStr[2000] = str[0];
		for(int i = 1; i < len; i++) {
			if(str[i] >= outputStr[start]) {
				outputStr[--start] = str[i];
			} else {
				outputStr[++end] = str[i];
			}
		}

		//output
		for(int i = start; i <= end; i++) {
			output << outputStr[i];
		}
		output << endl;
	}

	return 0;
}

